mods/other/heal_block/init.lua

```diff
@@ -0,0 +1,57 @@
+local last_heal_time = {}
+minetest.register_node("heal_block:heal", {
+    description = "Healing Block",
+    drawtype = "nodebox", 
+    node_box = { 
+        type = "fixed",
+        fixed = {-0.5, -0.5, -0.5, 0.5, 0, 0.5}
+    },
+    walkable = true,
+    tiles = {
+        {name = "heal_node.png", align_style = "repeat", scale = 1, position = {x = 0, y = 0, z = 0.5}}, -- top
+        {name = "default_snow.png", align_style = "repeat", scale = 1, position = {x = 0, y = 0, z = -0.5}}, -- bottom
+        {name = "default_snow.png", align_style = "repeat", scale = 1, position = {x = -0.5, y = 0, z = 0}}, -- left
+        {name = "default_snow.png", align_style = "repeat", scale = 1, position = {x = 0.5, y = 0, z = 0}}, -- right
+        {name = "default_snow.png", align_style = "repeat", scale = 1, position = {x = 0, y = -0.5, z = 0}}, -- front
+        {name = "default_snow.png", align_style = "repeat", scale = 1, position = {x = 0, y = 0.5, z = 0}} -- back
+    },
+    groups = {snappy=2,cracky=3,oddly_breakable_by_hand=3},
+    drop = "",
+    on_place = function(itemstack, placer, pointed_thing)
+        local pteam = ctf_teams.get(placer)
+        if pteam then
+            if not ctf_core.pos_inside(pointed_thing.under, ctf_teams.get_team_territory(pteam)) then
+                minetest.chat_send_player(placer:get_player_name(), "You can only place heal blocks in your own territory!")
+                return itemstack
+            end
+        end
+        minetest.item_place(itemstack, placer, pointed_thing)
+        return ItemStack{}
+    end,
+
+    on_timer = function(pos, elapsed)
+        local players_in_range = {}
+        for _, obj in ipairs(minetest.get_objects_inside_radius(pos, 3)) do
+            if obj:is_player() then
+                table.insert(players_in_range, obj)
+            end
+        end
+        for _, player in ipairs(players_in_range) do
+            local player_name = player:get_player_name()
+            if not last_heal_time[player_name] or os.time() - last_heal_time[player_name] >= 1 then
+                local hp = player:get_hp()
+                player:set_hp(hp + 1)
+                last_heal_time[player_name] = os.time()
+            end
+        end
+    end
+})
+
+minetest.register_abm({
```

Comment:
There is no need to use an ABM to initialize a node timer. ABM is a timer already, so it could trigger the required action directly (checking players in range and adding them HP). However, for nodes that don't occur frequently in the world (such as your heal block, versus e.g. a grass), a nodetimer should be used directly.

Looking up [here](https://rubenwardy.com/minetest_modding_book/en/items/callbacks.html), you can see how to use the on_construct node callback. Using that, you can initialize the node timer when it gets placed on the map.
Then [here](https://rubenwardy.com/minetest_modding_book/en/map/timers.html), you can see that returning true in the on_timer callback would cause the timer to run again for the same interval.

I suggest that you, in the free time, read the pages in this modding book. It would give you some ideas and suggestions for how to do things in Minetest, and then you can better see what approach best fits what situation.
It is organized into topics with concrete examples, unlike the pure API reference which is more abstract. The page for each topic also isn't too long, so it doesn't take up much time to read. : )